benches/dapol.rs

```
@@ -1,175 +1,385 @@
-        let mut rng = thread_rng();
-        let item_range = Uniform::new(0usize, num_leaves);
-
-        let dapol = build_dapol_tree::<blake3::Hasher, RangeProofSplitting>(&items, tree_height);
-        group.bench_function(BenchmarkId::new("splitting", tree_height), |bench| {
-            bench.iter_batched(
-                || {
-                    // generate a proof
-                    let item_idx = rng.sample(item_range);
-                    let tree_index = &items[item_idx].0;
-                    (item_idx, dapol.generate_proof(tree_index).unwrap())
-                },
-                |(item_idx, proof)| {
-                    // time proof verification
-                    proof.verify(&dapol.root(), &items[item_idx].1.get_proof_node())
-                },
-                BatchSize::SmallInput,
-            );
-        });
-
-        let dapol = build_dapol_tree::<blake3::Hasher, RangeProofPadding>(&items, tree_height);
-        group.bench_function(BenchmarkId::new("padding", tree_height), |bench| {
-            bench.iter_batched(
-                || {
-                    // generate a proof
-                    let item_idx = rng.sample(item_range);
-                    let tree_index = &items[item_idx].0;
-                    (item_idx, dapol.generate_proof(tree_index).unwrap())
-                },
-                |(item_idx, proof)| {
-                    // time proof verification
-                    proof.verify(&dapol.root(), &items[item_idx].1.get_proof_node())
-                },
-                BatchSize::SmallInput,
-            );
-        });
+                group.bench_function(
+                    BenchmarkId::new(
+                        "build_tree",
+                        format!("{:?}/{:?}/NUM_USERS: {:?}", &tup.0, &tup.1, &tup.2),
+                    ),
+                    |bench| {
+                        bench.iter(|| {
+                            ndm_smt = setup::build_ndm_smt(tup.clone()).ok();
```

This will just swallow any legit errors that happen during tree build. Not that we expect which any errors, but if there is an error for some reason we wouldn't know. Wrt errors you should take this approach:
- if there is an error in the API then fail the whole bench suite
- if the input tuple is not valid then continue the execution